pkg/apis/extensions/types.go

```diff
@@ -818,3 +819,68 @@ type PodSecurityPolicyList struct {
 
 	Items []PodSecurityPolicy `json:"items"`
 }
+
+// Template contains the inputs needed to produce a Config.
+type Template struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"`
+
+	Spec TemplateSpec `json:"spec,omitempty"`
+
+	Status TemplateStatus `json:"status,omitempty"`
+}
+
+type TemplateSpec struct {
+	// Optional: Parameters is an array of Parameters used during the
+	// Template to Config transformation.
+	Parameters []Parameter `json:"parameters,omitempty"`
+
+	// Required: A list of resources to create
+	Objects []runtime.Object `json:"objects,omitempty"`
```

Comment:
Wouldn't the substitutions be tons easier if you kept this as straight yaml or json instead of as runtime.Objects?